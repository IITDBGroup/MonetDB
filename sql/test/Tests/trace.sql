-- disable parallelism (mitosis & dataflow) to avoid ambiguous results 
set optimizer='sequential_pipe';

TRACE SELECT count(*) FROM types WHERE id < 30 AND systemname <> 'hge';
SELECT COUNT(*) FROM tracelog();
