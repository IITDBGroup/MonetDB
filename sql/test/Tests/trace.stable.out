stdout of test 'trace` in directory 'sql/test` itself:


# 12:44:31 >  
# 12:44:31 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33160" "--set" "mapi_usock=/var/tmp/mtest-9889/.s.monetdb.33160" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/dinther/INSTALL/var/MonetDB/mTests_sql_test" "--set" "embedded_r=yes" "--set" "embedded_py=2" "--set" "embedded_c=true"
# 12:44:31 >  

# MonetDB 5 server v11.34.0 (hg id: 8a60f0f74a17+)
# This is an unreleased version
# Serving database 'mTests_sql_test', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 31.305 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2019 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://catskill.da.cwi.nl:33160/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-9889/.s.monetdb.33160
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/Python2 module loaded
# MonetDB/R   module loaded

# 12:44:32 >  
# 12:44:32 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-9889" "--port=33160"
# 12:44:32 >  

#set optimizer='sequential_pipe';
#TRACE SELECT count(*) FROM types WHERE id < 30 AND systemname <> 'hge';
% sys.L3 # table_name
% L3 # name
% bigint # type
% 2 # length
[ 26	]
#TRACE SELECT count(*) FROM types WHERE id < 30 AND systemname <> 'hge';
% .trace,	.trace # table_name
% usec,	statement # name
% bigint,	clob # type
% 3,	207 # length
[ 86,	"X_3=0@0:void := querylog.define(\"trace select count(*) from types where id < 30 and systemname <> \\\\'hge\\\\';\":str, \"sequential_pipe\":str, 25:int);"	]
[ 44,	"X_6=0:int := sql.mvc();"	]
[ 155,	"X_20=<tmp_146>[35]:bat[:str] := sql.bind(X_6=0:int, \"sys\":str, \"types\":str, \"systemname\":str, 0:int);"	]
[ 78,	"X_10=<tmp_224>[35]:bat[:int] := sql.bind(X_6=0:int, \"sys\":str, \"types\":str, \"id\":str, 0:int);"	]
[ 71,	"C_7=<tmp_2330>[35]:bat[:oid] := sql.tid(X_6=0:int, \"sys\":str, \"types\":str);"	]
[ 112,	"C_88=<tmp_3431>[28]:bat[:oid] := algebra.thetaselect(X_10=<tmp_224>[35]:bat[:int], C_7=<tmp_2330>[35]:bat[:oid], 30:int, \"<\":str);"	]
[ 86,	"(X_15=<sql_empty_oid_bat>[0]:bat[:oid], X_16=<sql_empty_int_bat>[0]:bat[:int]) := sql.bind(X_6=0:int, \"sys\":str, \"types\":str, \"id\":str, 2:int);"	]
[ 86,	"C_89=<tmp_3113>[0]:bat[:oid] := algebra.thetaselect(X_16=<sql_empty_int_bat>[0]:bat[:int], nil:bat[:oid], 30:int, \"<\":str);"	]
[ 75,	"X_13=<sql_empty_int_bat>[0]:bat[:int] := sql.bind(X_6=0:int, \"sys\":str, \"types\":str, \"id\":str, 1:int);"	]
[ 85,	"C_91=<tmp_2750>[0]:bat[:oid] := algebra.thetaselect(X_13=<sql_empty_int_bat>[0]:bat[:int], C_7=<tmp_2330>[35]:bat[:oid], 30:int, \"<\":str);"	]
[ 96,	"C_29=<tmp_3431>[28]:bat[:oid] := sql.subdelta(C_88=<tmp_3431>[28]:bat[:oid], C_7=<tmp_2330>[35]:bat[:oid], X_15=<sql_empty_oid_bat>[0]:bat[:oid], C_89=<tmp_3113>[0]:bat[:oid], C_91=<tmp_2750>[0]:bat[:oid]);"	]
[ 87,	"C_92=<tmp_2750>[26]:bat[:oid] := algebra.thetaselect(X_20=<tmp_146>[35]:bat[:str], C_29=<tmp_3431>[28]:bat[:oid], \"hge\":str, \"!=\":str);"	]
[ 88,	"(X_23=<sql_empty_oid_bat>[0]:bat[:oid], X_24=<sql_empty_str_bat>[0]:bat[:str]) := sql.bind(X_6=0:int, \"sys\":str, \"types\":str, \"systemname\":str, 2:int);"	]
[ 75,	"C_93=<tmp_3113>[0]:bat[:oid] := algebra.thetaselect(X_24=<sql_empty_str_bat>[0]:bat[:str], nil:bat[:oid], \"hge\":str, \"!=\":str);"	]
[ 77,	"X_22=<sql_empty_str_bat>[0]:bat[:str] := sql.bind(X_6=0:int, \"sys\":str, \"types\":str, \"systemname\":str, 1:int);"	]
[ 79,	"C_95=<tmp_2330>[0]:bat[:oid] := algebra.thetaselect(X_22=<sql_empty_str_bat>[0]:bat[:str], C_29=<tmp_3431>[28]:bat[:oid], \"hge\":str, \"!=\":str);"	]
[ 94,	"C_36=<tmp_2750>[26]:bat[:oid] := sql.subdelta(C_92=<tmp_2750>[26]:bat[:oid], C_29=<tmp_3431>[28]:bat[:oid], X_23=<sql_empty_oid_bat>[0]:bat[:oid], C_93=<tmp_3113>[0]:bat[:oid], C_95=<tmp_2330>[0]:bat[:oid]);"	]
[ 72,	"X_40=26:lng := aggr.count(C_36=<tmp_2750>[26]:bat[:oid]);"	]
[ 152,	"sql.resultSet(\"sys.L3\":str, \"L3\":str, \"bigint\":str, 64:int, 0:int, 7:int, X_40=26:lng);"	]
#SELECT COUNT(*) FROM tracelog();
% .L4 # table_name
% L4 # name
% bigint # type
% 2 # length
[ 19	]

# 12:44:32 >  
# 12:44:32 >  "Done."
# 12:44:32 >  

